actors/evm/src/interpreter/instructions/call.rs

```diff
@@ -260,26 +263,40 @@ pub fn call_generic<RT: Runtime>(
                         )
                     }
                 }
-                CallKind::DelegateCall => {
-                    // first invoke GetBytecode to get the code CID from the target
-                    let code = crate::interpreter::instructions::ext::get_evm_bytecode_cid(
-                        system.rt, dst,
-                    )?;
-
-                    // and then invoke self with delegate; readonly context is sticky
-                    let params = DelegateCallParams {
-                        code,
-                        input: input_data.to_vec(),
-                        readonly: system.readonly,
-                    };
-                    system.send(
-                        &system.rt.message().receiver(),
-                        Method::InvokeContractDelegate as u64,
-                        RawBytes::serialize(&params)?,
-                        TokenAmount::from(&value),
-                    )
-                }
+                CallKind::DelegateCall => match get_cid_type(system.rt, dst) {
```

Comment:
I removed the closure/return. We didn't need them.

Generally speaking, jumping like that is a "last resort" kind of thing as it can make code really hard to follow. It also increases diffs by changing indent.

Reply:
yeah diff was gross for sure. In all honesty this method ended up a lot cleaner than i was thinking (idk what i was thinking tbh). I probably should've given it at least a few minutes of poking before dismissing it as unfeasible.